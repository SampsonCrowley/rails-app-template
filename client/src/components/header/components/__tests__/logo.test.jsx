import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import Logo from '../logo'

describe('Components - Header::Logo', () => {
  const div = document.createElement('div');

  const createLogo = ({...props}) => {
    ReactDOM.render((
      <Router>
        <Logo {...props} />
      </Router>
    ), div);
    return div.querySelector('div.header-logo')
  }

  it('renders the header-logo', () => {
    const rendered = createLogo()
    expect(rendered).toBeTruthy()
    expect(rendered.tagName).toBe("DIV")
    expect(rendered.classList.contains('header-logo')).toBeTruthy()
    ReactDOM.unmountComponentAtNode(div);
  })

  it('contains a brand image', () => {
    const rendered = createLogo().querySelector('img')
    expect(rendered).toBeTruthy()
    expect(rendered.tagName).toBe("IMG")
    expect(rendered.alt).toBe('Site Logo')
    ReactDOM.unmountComponentAtNode(div);
  })

  it('logo is wrapped in a link home', () => {
    const base = createLogo(),
          img = base.querySelector('a > img'),
          linked = base.querySelector('a')
    expect(img).toBeTruthy()
    expect(linked.href).toMatch(/http:\/{2}[A-Za-z09\.]+\/?$/)
    ReactDOM.unmountComponentAtNode(div);
  })

  it('is snapshotable', () => {
    const tree = renderer
      .create(
        <Router>
          <Logo />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
  })
})
