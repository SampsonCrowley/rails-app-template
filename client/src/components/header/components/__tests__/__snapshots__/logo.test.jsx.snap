// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Header::Logo is snapshotable 1`] = `
<div
  className="header-logo"
>
  <a
    className="header-link"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Site Logo"
      src="dus-logo.png"
    />
  </a>
</div>
`;
